source/Halibut/Transport/SecureListener.cs

```diff
@@ -200,7 +200,17 @@ async Task ExecuteRequest(TcpClient client)
 
                     if (Authorize(ssl, clientName))
                     {
+                        // The ExchangeMessage call can hang on reading the stream which keeps a thread alive,
+                        // so we dispose the stream which will cause the thread to abort with an exceptions.
+                        var weakSSL = new WeakReference(ssl);
+                        cts.Token.Register(() =>
+                        {
+                            if (weakSSL.IsAlive)
+                                ((IDisposable)weakSSL.Target).Dispose();
+                        }, true);
+
                         // Delegate the open stream to the protocol handler - we no longer own the stream lifetime
+                        //await Task.WhenAny(cts.Token.WhenCanceled(), ExchangeMessages(ssl));
```

Comment:
Not needed?

Reply:
Whoops. Leftover from other things I was trying.